### DISCUSSION TITLE
stability booklet

### DISCUSSION THREAD

**Discussion by githubzhangfei** [dragonflydb/dragonfly#2935] [Q&A]
When I queried Dragonfly db, the performance was good most of the time. But occasionally there will be a query that takes 30 milliseconds or so. Why is that? It's the same batch of data. Large deviations occur after multiple executions

**Comment by adiholden** [comment]
Hi @githubzhangfei 
how do you run dragonfly? can you share INFO ALL output?
do you have a test code I can reproduces the problem?

**Reply by githubzhangfei** [reply]
JedisPool jedisPool = new JedisPool(poolConfig, "192.168.10.111", 8080, 1000, null, 0);
        for (int i = 0; i < 20000000; i++) {
            Thread.sleep(5);
            Jedis jedis = jedisPool.getResource();
            Pipeline pipelined = jedis.pipelined();
            for (int j = 0; j < 61; j++) {
                pipelined.get("0" + i + UUID.randomUUID());
            }
            long start = System.currentTimeMillis();
            pipelined.syncAndReturnAll();
            if ((System.currentTimeMillis() - start) > 20) {
                System.out.println("time out 20ms");
            }
            jedis.close();
        }
The code looks like this, but in practice it's running at around 5,000 per second, using a pipeline of around 60 get commands per query. Dragonfly db over 20mm accounted for about 6 per thousand. What I want to know is why this happened. My data volume is about 50 million. The data is static, which I understand should not be the case. Does he have a mechanism? I deployed Dragonfly db with docker as the master and slave, only Dragonfly db was installed on the server, and the server configuration was very high. And please help me.

**Comment by adiholden** [comment]
sorry @githubzhangfei for the late response. can you please share info all output? you should not get this delays
